components/num-util/src/fixed_decimal.rs

```diff
@@ -519,3 +699,31 @@ fn test_lower_magnitude_bounds() {
     assert_eq!(Error::Limit, dec.multiply_pow10(-1).unwrap_err());
     assert_eq!(dec, dec_backup, "Value should be unchanged on failure");
 }
+
+#[test]
+fn test_from_str_isize_limits() {
```

Comment:
Probably merge these two test cases into `test_isize_limits` and `test_ui128_limits`